openedx/core/djangoapps/content_tagging/types.py

```diff
@@ -1,8 +1,26 @@
 """
 Types used by content tagging API and implementation
 """
-from typing import Union
+from __future__ import annotations
+from typing import Union, Dict, List
 
+from attrs import define
 from opaque_keys.edx.keys import LearningContextKey, UsageKey
+from openedx_tagging.core.tagging.models import ObjectTag, Taxonomy
+from xblock.core import XBlock
 
 ContentKey = Union[LearningContextKey, UsageKey]
+
+ObjectTagByTaxonomyIdDict = Dict[int, List[ObjectTag]]
+ObjectTagByObjectIdDict = Dict[str, ObjectTagByTaxonomyIdDict]
```

Comment:
```suggestion
ObjectTagsByTaxonomyId = dict[int, list[ObjectTag]]
ObjectTagsByObjectId = dict[ContentKey, ObjectTagsByTaxonomyId]
```

Just use `dict` and `list` - you don't need to import from `types` anymore. And if you can use `ContentKey` instead of `str` that's better since it's more explicit, but don't bother if it makes the code messier.

Reply:
This works with params, but using for declare types throws `error: "dict" is not subscriptable  [misc]`.

We have the `type` statement in 3.12 for type alias, but I don't think we can use it here..
https://docs.python.org/3/reference/simple_stmts.html#the-type-statement

Reply:
Ah, right. That's also why we're using `Union` above in the part I wrote. I should have remembered that.